src/Components/Hints.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react";
+import { BsCircleFill, BsXCircle, BsCircle } from "react-icons/bs";
+import { constraints } from "../constants";
+
+const Hints = () => {
+  const [show, setShow] = useState<boolean>(false);
```

Comment:
better naming would be isVisible

Reply:
changed the name to [isVisible, setVisible]